src/aiodynamo/errors.py

```diff
@@ -5,6 +5,10 @@ class AIODynamoError(Exception):
     pass
 
 
+class NoCredentialsFound(Exception):
```

Comment:
I'm thinking about pros/cons of inheriting from `Exception` and not `AIODynamoError`...

I wonder what's least unexpected:
* first failed request almost killing user program
* or each request failing with same error

My weak preference would be for the new exception to inherit from `AIODynamoError` because:
1. user program may be using several libraries together, and `aiodynamo` should not be "special".
2. if a user program only uses `aiodynamo` but wants to exit quickly if there are no credentials, it can still catch this type

Reply:
> I'm thinking about pros/cons of inheriting from `Exception` and not `AIODynamoError`...

That was a complete brainfart from me, I intended to inherit from the root exception but for some reason wrote `Exception`.